lib/http.js

```diff
@@ -23,11 +28,22 @@ const client = require('_http_client');
 const ClientRequest = exports.ClientRequest = client.ClientRequest;
 
 exports.request = function request(options, cb) {
+  if (typeof options === 'string') {
+    options = url.parse(options);
+    if (!options.hostname) {
+      throw new Error('Unable to determine the domain name');
+    }
+  } else if (options instanceof url.URL) {
```

Comment:
@mscdex ... you could export the `context` symbol from `internal/url` and check for that. All `URL` instances would have that set internally.